src/System.Net.Requests/src/System/Net/WebExceptionPal.Unix.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+
+namespace System.Net
+{
+    public partial class WebException : InvalidOperationException
+    {
+        internal static WebExceptionStatus GetStatusFromException(HttpRequestException ex)
+        {
+            WebExceptionStatus status;
+
+            // Issue 2384: update WebException.GetStatusFromException after System.Net.Http API changes
+            //
+            // For now, we use the .HResult of the exception to help us map to a suitable
+            // WebExceptionStatus enum value.  The .HResult is set into this exception by
+            // the underlying .NET Core and .NET Native versions of the System.Net.Http stack.
+            // In the future, the HttpRequestException will have its own .Status property that is
+            // an enum type that is more compatible directly with the WebExceptionStatus enum.
+            switch (ex.HResult)
+            {
+                case Interop.libcurl.CURLcode.CURLE_COULDNT_RESOLVE_HOST:
```

Comment:
As part of preserving app-compat, we try to replicate the same error codes into the WebException.Status property.  This won't always work because HttpWebRequest is no longer based on the original managed sockets HTTP stack, but rather built on top of HttpClient which is built on top of platform-specific HTTP stacks.

We reviewed top apps to determine which WebExceptionStatus enum values were the most important to try to replicate.  I am still in the process of doing some additional work in this area for this contract. And as @kapilash points out, the error/exception mechanism is being altered more generally for HttpClient since that API surface is the more modern one going forward; whereas System.Net.Requests is considered a 'legacy' API surface.